### ISSUE TITLE
A document for the best practice in setting up a development environment for testing

### ISSUE THREAD

**Issue by jonathanbossenger** [wp-cli/handbook#10]
I'm happy to start this document, but I need some guidance.

The idea being a 'how to setup wp-cil on a development environment for a first timer'

Considering things like LAMP. MAMP, WAMP, Vagrant (and possibly Docker) in terms of what people use for development environments.

**Comment by danielbachhuber** [comment]
@jonathanbossenger I think this might make most sense as a couple sentences in the [Setting up](https://github.com/wp-cli/handbook/blob/master/pull-requests.md#setting-up) section of our pull requests documentation. Would you like to contribute a change to that effect?

**Comment by jonathanbossenger** [comment]
@danielbachhuber certainly. As previously mentioned though, I've not yet (probably due to lack of sufficient time) successfully setup a development environment. I guess what I'm asking is that would you or @schlessera mind if I ping you in Slack should I get stuck?

**Comment by danielbachhuber** [comment]
> I guess what I'm asking is that would you or @schlessera mind if I ping you in Slack should I get stuck?

Not at all. `#cli` is a good channel for any questions because you'll be able to reach whoever is around at the time.

> As previously mentioned though, I've not yet (probably due to lack of sufficient time) successfully setup a development environment.

For WP-CLI specifically or WordPress generally? If the former, you should be able to reuse your existing WordPress development environment for WP-CLI development.
